query-graphs-js/src/graphPath.ts

```diff
@@ -387,7 +387,10 @@ export class GraphPath<TTrigger, RV extends Vertex = Vertex, TNullEdge extends n
    * The set of edges that may legally continue this path.
    */
   nextEdges(): readonly Edge[] {
-    return this.graph.outEdges(this.tail);
+    const tailEdge = this.edgeToTail;
+    return tailEdge
```

Comment:
If this is an instance of `TNullEdge`, is it guaranteed to be false here? I don't know how to interpret a type that extends null.

Reply:
My understanding/intent (and hopefully I didn't miss something) is that the only type extending `null` is `null` itself, so this is really a way to say that `tailEdge` can be only be one of an `Edge`, `null`, or `undefined`, which some specific instantiation making the `null` part guaranteed to never happen.

Here, we'll default to `outEdges()` if `tailEdge` is `null`, which is fine. That is, it's always safe to return `outEdges()` from this method, with `nonTrivialFollowupEdges()` being simply a potentially more efficient alternative, and `null` edges in a `GraphPath` are super rare (they can only happen during query planning, not validation composition, and only when the input query has some essentially useless type-condition that we keep in the path only because there is some execution directives associated we don't want to lose) so I really don't think it's worth doing extra work to try to optimise them in this case.

Anyway, I did added a comment that may help a bit with was is going on there.